src/libraries/icubmod/embObjLib/ethBoards.cpp

```diff
@@ -54,15 +56,31 @@
 
 eth::EthBoards::EthBoards()
 {
-    memset(LUT, 0, sizeof(LUT));
+    // memset(LUT, 0, sizeof(LUT));
```

Comment:
the `memset()` was a quick way to do initialization. If some compiler complains you can drop any code in the constructor and place initialization in the `ethBoards.h` file by improving the declaration of `class EthBoards` as follows:

```C++

        use member inialization

        struct ethboardProperties_t
        {
            eOipv4addr_t            ipv4 {0};
            std::string             name {""};
            uint8_t                 numberofinterfaces {0};
            uint8_t                 boardnumber {0};
            AbstractEthResource*    resource {nullptr};
            IethResource*           interfaces[iethresType_numberof] {nullptr};
            void clear() 
            { 
                // do what you need 
            }
        };

 and also ...

        size_t sizeofLUT {0};
        ethboardProperties_t LUT[EthBoards::maxEthBoards] {};

```

with the above code you don't need anything in the ctor
```C++
eth::EthBoards::EthBoards()
{
    // nothing is required because LUT and sizeofLUT are already OK. 
    // BUT i just do the following anyway:
   for (auto& l : LUT)
   { 
       l.clear();
   }
    sizeofLUT = 0;
}
``` 

One more thing: 

you may also change `bool eth::EthBoards::rem(eth::AbstractEthResource* res)` to call
`LUT[index].clear();` instead of these [lines](https://github.com/robotology/icub-main/blob/devel/src/libraries/icubmod/embObjLib/ethBoards.cpp#L205-L213).

Reply:
Ok, I'll check this, update and see if it is still fine the execution of the code in Win